src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -1254,20 +1349,72 @@ public void OutputCode(IEnumerable<DependencyNode> nodes, MethodDesc entrypoint,
             Out.Write(_gcStatics.ToString());
             Out.Write("} __gcThreadStatics;");
 
-            OutputExternCSignatures();
 
-            foreach (var node in nodes)
+            foreach (var externC in _externCSignatureMap)
             {
-                if (node is CppMethodCodeNode)
-                    OutputMethodNode(node as CppMethodCodeNode);
+                string importName = externC.Key;
+                // TODO: hacky special-case
+                if (importName != "memmove" && importName != "malloc") // some methods are already declared by the CRT headers
+                {
+                    sb.AppendLine();
+                    sb.Append(GetCppMethodDeclaration(null, false, importName, externC.Value));
+                }
             }
+            Out.Write(sb.ToString());
+            sb.Clear();
+
+            Out.Write(methodImplementations.ToString());
+            methodImplementations.Clear();
 
             if (entrypoint != null)
             {
-                OutputMainMethodStub(entrypoint);
-            }
+                // Stub for main method
```

Comment:
I have moved this code to separate `OutputMainMethodStub` to make it better factored. Could you please keep it in separate function?